sympy/printing/numpy.py

```diff
@@ -0,0 +1,494 @@
+from sympy.core import S
+from .pycode import PythonCodePrinter, _known_functions_math, _print_known_const, _print_known_func, _unpack_integral_limits
+from .codeprinter import CodePrinter
+
+
+_not_in_numpy = 'erf erfc factorial gamma loggamma'.split()
+_in_numpy = [(k, v) for k, v in _known_functions_math.items() if k not in _not_in_numpy]
+_known_functions_numpy = dict(_in_numpy, **{
+    'acos': 'arccos',
+    'acosh': 'arccosh',
+    'asin': 'arcsin',
+    'asinh': 'arcsinh',
+    'atan': 'arctan',
+    'atan2': 'arctan2',
+    'atanh': 'arctanh',
+    'exp2': 'exp2',
+    'sign': 'sign',
+    'logaddexp': 'logaddexp',
+    'logaddexp2': 'logaddexp2',
+})
+_known_constants_numpy = {
+    'Exp1': 'e',
+    'Pi': 'pi',
+    'EulerGamma': 'euler_gamma',
+    'NaN': 'nan',
+    'Infinity': 'PINF',
+    'NegativeInfinity': 'NINF'
+}
+
+
+class NumPyPrinter(PythonCodePrinter):
+    """
+    Numpy printer which handles vectorized piecewise functions,
+    logical operators, etc.
+    """
+
+    _module = 'NumPy'
```

Comment:
Shouldn't this be lowercase?

Reply:
I wanted to preserve the formatting of "NumPy" for the `self.language = "Python with {}".format(self._module)` line further down. You may notice that I call `.lower()` on `_module` a few times, and then overwrite it to be lowercase. 
The following lines, which are called _after_ the `class NumPyPrinter...`, made writing the code rather tricky. If they can be called via the `def __init__()` instead, then I could clean up some parts.
```
# line 315
for k in NumPyPrinter._kf:
    setattr(NumPyPrinter, '_print_%s' % k, _print_known_func)

for k in NumPyPrinter._kc:
    setattr(NumPyPrinter, '_print_%s' % k, _print_known_const)
```

Reply:
I find that to be very confusing. Where is `self.language` used? I would just make numpy lowercase everywhere.

Reply:
I took the approach of not wanting to change too much unnecessarily out of fear of breaking / stepping on toes. In particular, there were several tests failing with NumPy -> numpy.
I've now edited those tests to be a bit simpler, and gone lowercase everywhere.